### PR TITLE
B-tagging WP SFs and systematics

### PR DESCRIPTION
**Pull request by andrea21z**
Changes:
- I have included the WP loose in the implematation of the b-tagging SFs method (following the [recommendations](https://twiki.cern.ch/twiki/bin/viewauth/CMS/BTagSFMethods)).
- Before we had only two syst. variations (btagSFUp and btagSFDown), and now they are splited by flavor (bc or light) and by correlated/uncorrelated:
        - btagSFbc_corrUp/Down
        - btagSFlight_corrUp/Down
        - btagSFbc_uncorrUp/Down
        - btagSFlight_uncorrUp/Down

I have checked the Data/MC agreement in the ttbar CR and the syst. variations and everything looks fine!

### CODE REVIEW

File: analysis/topEFT/topeft.py [hunk 1]
```diff
@@ -132,6 +133,7 @@ def process(self, events):
             sow_factDown       = self._samples[dataset]["nSumOfWeights_factDown"]
             sow_renormfactUp   = self._samples[dataset]["nSumOfWeights_renormfactUp"]
             sow_renormfactDown = self._samples[dataset]["nSumOfWeights_renormfactDown"]
+        '''
```
Comment (bryates) [inline analysis/topEFT/topeft.py:136]: Can you remove the block comments here?

**Review by bryates — CHANGES_REQUESTED** [review]
Thanks @andrea21z, I just had one change. I'll update the datacard to handle the correlations, but we don't need to hold off on this PR.

File: analysis/topEFT/topeft.py (hunk 1 above)
Comment (andrea21z) [reply analysis/topEFT/topeft.py:136]: yes, I  forget to uncomment it

**Review by bryates — APPROVED** [review]
Looks good!

**Comment by andrea21z** [comment]
> @andrea21z looks like the datacard checks are failing. Do we expect these updates to affect the nominal yields?

Yes, they will change a little

**Comment by bryates** [comment]
> > @andrea21z looks like the datacard checks are failing. Do we expect these updates to affect the nominal yields?
> 
> Yes, they will change a little

Ok, then can you update the reference files? Once that is done, I think this PR should be ready to merge.

**Comment by bryates** [comment]
> > > > @andrea21z looks like the datacard checks are failing. Do we expect these updates to affect the nominal yields?
> > > 
> > > 
> > > Yes, they will change a little
> > 
> > 
> > Ok, then can you update the reference files? Once that is done, I think this PR should be ready to merge.
> 
> I have updated the reference json file, but I think I can't update the datacard files, those files were always updated by Kelci or you because I can't run those tests in Oviedo (or maybe other reason, I don't remember...)

Ok, I can't remember either, but I'll take care of it now.

**Comment by andrea21z** [comment]
> > > > > @andrea21z looks like the datacard checks are failing. Do we expect these updates to affect the nominal yields?
> > > > 
> > > > 
> > > > Yes, they will change a little
> > > 
> > > 
> > > Ok, then can you update the reference files? Once that is done, I think this PR should be ready to merge.
> > 
> > 
> > I have updated the reference json file, but I think I can't update the datacard files, those files were always updated by Kelci or you because I can't run those tests in Oviedo (or maybe other reason, I don't remember...)
> 
> Ok, I can't remember either, but I'll take care of it now.

Thank you very much :)

**Comment by kmohrman** [comment]
Thanks @andrea21z, this PR looks good to me. I just pushed two small commits:
* Update ref yields (the ones for the larger check than the CI check)
* Revert remorm/fact stuff to match the master branch (we do not include renorm and fact right now, and instead just use renormfact since we're not currently doing the envelope approach)

**Pull request merged**